tools/api-server/server.js

```diff
@@ -79,49 +79,70 @@ app.get('/topics_projects', cors(), function (req, res) {
 
 
 app.get('/taxonomy', cors(), function (req, res) {
-	query = "select category, display_name, id, synonym"
+	query = "select category, subcategory, display_name, id, synonym"
 		  + " from taxonomy_tags_synonyms";
 
 	category = req.query.category;
 	if (category != undefined) {
 		query += " where category = '" + category + "'";
 	}
 
-	query += " order by category, display_name, synonym";
-	//query += " limit 10";
+	query += " order by category, subcategory, display_name, synonym";
 
 	console.log('query: ' + query);
 	pool.query(query, function (err, rows, fields) {
 	  if (err) throw err
 
 	  var cat = "";
+	  var subcat = "";
+	  var subcateg = null;
 	  var categories = new Array();
+	  var subcategories = new Array();
 	  var c = 0;
-	  var it = "";
+	  var sc = 0;
+	  var item_name = "";
 	  for(var row of rows) {
-		  //console.log("Row: " + row.display_name);
+		  //console.log("Row: " + row.display_name + " " + row.category + " " + row.subcategory);
 		  if(cat != row.category) {
+			  subcat = "";
+			  subcateg = null;
 			  cat = row.category;
+			  //console.log("New category: " + cat);
 			  categ = new Object();
 			  categ.text = cat;
 			  categories[c] = categ;
 			  categ.children = new Array();
 			  c = c+1;
-			  it = "";
+			  item_name = "";
 		  }
-		  if(it != row.display_name) {
-			it = row.display_name;
+		  if(row.subcategory != null && subcat != row.subcategory) {
+			  subcat = row.subcategory;
+			  //console.log("New subcategory: " + subcat);
+			  subcateg = new Object();
+			  subcateg.text = subcat;
+			  subcategories[sc] = categ;
+			  subcateg.children = new Array();
```

Comment:
Unless it changes the logic, perhaps simplify to 

```suggestion
			  subcateg = { text: subcat, children: [] };
			  subcategories[sc] = categ;
```

and remove `subcategories` if it isn't used